### ISSUE TITLE
Harmonize tick-resolution-choosing and tick-format-choosing logic

### ISSUE THREAD

**Issue by spencerkclark** [SciTools/nc-time-axis#48]
[The code that chooses the resolution of the ticks](https://github.com/SciTools/nc-time-axis/blob/531dd0da83f2ea6b351f129559ccd4e8b0ccbb5f/nc_time_axis/__init__.py#L109-L134) and [the code that chooses the format of the ticks](https://github.com/SciTools/nc-time-axis/blob/531dd0da83f2ea6b351f129559ccd4e8b0ccbb5f/nc_time_axis/__init__.py#L55-L80) are sometimes not consistent with each other.  This can lead to confusing axis tick labels, e.g. in the following example provided by @klindsay28: 

```python
import cftime
import matplotlib.pyplot as plt
import nc_time_axis

time_vals = [cftime.DatetimeNoLeap(1+year, 1+month, 15)
             for year in range(3) for month in range(12)]

x_vals = [time_val.year + time_val.dayofyr / 365.0
          for time_val in time_vals]

fig, ax = plt.subplots(1, 1)
ax.plot(time_vals, x_vals, "-o")
```

![image](https://user-images.githubusercontent.com/6628425/92311612-8688f480-ef86-11ea-937e-6bffd9f38b18.png)

In this case a monthly resolution is chosen for the ticks, but a tick format of `"%Y"` is used, giving the false impression that the ticks represent the beginning of years.  I would expect for monthly resolution ticks, a format of `"%Y-%m"` would be used.  In this example this would result in a plot that looked like this:

![updated](https://user-images.githubusercontent.com/6628425/92311656-e1bae700-ef86-11ea-9f00-70eb9e42e3c4.png)

If possible it would be great if we could make sure that inconsistencies like the above example could be avoided.

xref: https://github.com/pydata/xarray/issues/4401

**Comment by bjlittle** [comment]
@spencerkclark Is this still a live issue that we want to pull into `v1.3.0` ?

**Comment by spencerkclark** [comment]
Yes, this is still a live issue and would be great to fix someday.  I think it can wait until a future `nc-time-axis` release, however, since it involves a little more than just catching up with changes in upstream packages.

**Comment by bjlittle** [comment]
@spencerkclark Awesome. Okay, agreed :+1: 

I'm going to create a `1.4.0` milestone, project and GH discussion, so that we can easily triage all such issues and PRs, and it'll help slowly shape the content of the next, next release :wink:

**Comment by spencerkclark** [comment]
Closed by #79.
